authentication/views.py

```diff
@@ -0,0 +1,137 @@
+import os
+
+import google_auth_oauthlib.flow
+from django.contrib.auth import user_logged_in
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.decorators import permission_classes
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+
+from authentication.models import User
+from .apps import scopes, Request
+from .tokens import gen_tokens, sendTokens, get_tokens_for_user
+
+
+def get_url(flow, **kwargs):
+    authorization_url, state = flow.authorization_url(access_type='offline',
+                                                      include_granted_scopes='true',
+                                                      **kwargs)
+    return authorization_url, state
+
+
+"""
+This API VIEW will handle the user authentication if user want to get an refresh token then pass prompt='consent' 
+in the url params. after generating the url, user will be redirected to the url where all the authentication is done
+"""
+
+
+@api_view(['POST'])
+@permission_classes([AllowAny])
+def google_sign_in(request: Request):
```

Comment:
Let's convert all these function-based views to class-based views sub-classed from appropriate classes 
in the DjangoRestFramework documentation.

Reply:
done